static/app/components/profiling/flamegraph/flamegraphPreview.tsx

```diff
@@ -0,0 +1,244 @@
+import styled from '@emotion/styled';
+import {vec2} from 'gl-matrix';
+
+import ConfigStore from 'sentry/stores/configStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {defined} from 'sentry/utils';
+import {
+  CanvasPoolManager,
+  useCanvasScheduler,
+} from 'sentry/utils/profiling/canvasScheduler';
+import {CanvasView} from 'sentry/utils/profiling/canvasView';
+import {Flamegraph as FlamegraphModel} from 'sentry/utils/profiling/flamegraph';
+import {
+  DarkFlamegraphTheme,
+  LightFlamegraphTheme,
+} from 'sentry/utils/profiling/flamegraph/flamegraphTheme';
+import {FlamegraphCanvas} from 'sentry/utils/profiling/flamegraphCanvas';
+import {FlamegraphFrame} from 'sentry/utils/profiling/flamegraphFrame';
+import {Rect, useResizeCanvasObserver} from 'sentry/utils/profiling/gl/utils';
+import {FlamegraphRenderer2D} from 'sentry/utils/profiling/renderers/flamegraphRenderer2D';
+import {FlamegraphTextRenderer} from 'sentry/utils/profiling/renderers/flamegraphTextRenderer';
+import {useProfileGroup} from 'sentry/views/profiling/profileGroupProvider';
+
+interface FlamegraphPreviewProps {
+  relativeStartTimestamp: number;
+  relativeStopTimestamp: number;
+}
+
+export function FlamegraphPreview({
+  relativeStartTimestamp,
+  relativeStopTimestamp,
+}: FlamegraphPreviewProps) {
+  const canvasPoolManager = useMemo(() => new CanvasPoolManager(), []);
+  const scheduler = useCanvasScheduler(canvasPoolManager);
+
+  const {theme} = useLegacyStore(ConfigStore);
+  const flamegraphTheme = theme === 'light' ? LightFlamegraphTheme : DarkFlamegraphTheme;
+  const profileGroup = useProfileGroup();
+
+  const threadId = useMemo(
+    () => profileGroup.profiles[profileGroup.activeProfileIndex]?.threadId,
+    [profileGroup]
+  );
+
+  const profile = useMemo(() => {
+    if (!defined(threadId)) {
+      return null;
+    }
+    return profileGroup.profiles.find(p => p.threadId === threadId) ?? null;
+  }, [profileGroup.profiles, threadId]);
+
+  const flamegraph = useMemo(() => {
+    if (!defined(threadId) || !defined(profile)) {
+      return FlamegraphModel.Empty();
+    }
+
+    return new FlamegraphModel(profile, threadId, {});
+  }, [profile, threadId]);
+
+  const [flamegraphCanvasRef, setFlamegraphCanvasRef] =
+    useState<HTMLCanvasElement | null>(null);
+
+  const flamegraphCanvas = useMemo(() => {
+    if (!flamegraphCanvasRef) {
+      return null;
+    }
+    return new FlamegraphCanvas(flamegraphCanvasRef, vec2.fromValues(0, 0));
+  }, [flamegraphCanvasRef]);
+
+  const flamegraphView = useMemo(() => {
+    if (!flamegraphCanvas) {
+      return null;
+    }
+
+    const canvasView = new CanvasView({
+      canvas: flamegraphCanvas,
+      model: flamegraph,
+      options: {barHeight: flamegraphTheme.SIZES.BAR_HEIGHT},
+      mode: 'anchorBottom',
+    });
+
+    canvasView.setConfigView(
+      computePreviewConfigView(
+        flamegraph,
+        canvasView.configView,
+        relativeStartTimestamp * 1e9,
```

Comment:
We should multiply with profile.unit here I think (probably doesnt make a difference because we always use ns, but it would be more future proof in case at some point in the future we dont)